design.md

```diff
@@ -1,13 +1,25 @@
 
 ## Why did you choose this design, i.e., why did you think it would be a good idea for users to express the maze-searching computation using this syntax?
+I think there are two major changes that really help make writing programs easier in my design. First, states are named, not numbered. This helps the programmer keep track of what that state is supposed to accomplish or what they know is true in that state. Second, NEWS is replaced with up / right / down / left, which is more natural for the problem.
+
+For better or worse, I tried to make the syntax more natural so that it is easier for the programmer to express their thoughts and for a reader to determine the programmer's intentions. While natural language programming is generally a bad idea, I think this syntax is unambiguous and much easier to understand than before. This should help reduce the cognitive load of converting between ideas in your head and code. In fact, I think the syntax may even make most of the comments unnecessary as it flows really well.
+
+In later iterations of my design, I also tried to remove repeated elements out of the code by modifying the design itself. For instance, it makes a lot of sense to define rules starting in a particular state next to each other, so I moved the starting state out of the rules and grouped them by their starting state. I also heavily rely on implicit behavior to prevent the natural language code from getting too long. Most rules can be written in less than 80 characters without much effort at all. In fact, the longest line of code (excluding comments) is 69 characters in example-ideal.txt and 52 characters in example-ideal-maze.txt.
 
 ## What behaviors are easier to express in your design than in Picobot’s original design?  If there are no such behaviors, why not?
+The natural language syntax makes handling states, wildcards, not moving, and staying in the same state really easy because you don't really have to express them at all. Each different state name is implicity a different state number, so the programmer doesn't have to keep track of which number state does what (since there is no "state 0", the starting state is the first one in the code). Wildcards are automatically generated when the programmer's logic does not depend on whether they can move in a direction or not, don't moves are automatically generated when the programmer doesn't specify a movement direction, and transitions to the same state are automatically generated if the programmer doesn't specify a state transition. This helps keep programs short (despite the verbosity of natural language) and helps the programmer focus on the trickier parts of the implementation.
 
```

Comment:
We made the same design decision, but one question we wrestled with was, "Is that really easier?" Might people make a mistake in their code, like forgetting to put a move direction, and then have trouble debugging because the code still compiles and runs just fine so they don't realize they made a mistake?